packages/dicomImageLoader/src/imageLoader/isColorConversionRequirementsFulfilled.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * This function checks color space conversion data requirements before
+ * apply them. This function was created to solve problems like the one
+ * discussed in here https://discourse.orthanc-server.org/t/orthanc-convert-ybr-to-rgb-but-does-not-change-metadata/3533/17
+ * In this case, Orthanc server convertes the pixel data from YBR to RGB, but maintain
+ * the photometricInterpretation dicom tag in YBR
+ * @param imageFrame
+ * @param RGBA
+ * @returns
+ */
+export default function isColorConversionRequirementsFulfilled(
+  imageFrame,
+  RGBA
+) {
+  if (imageFrame === undefined) {
+    return false;
+  }
+  const { rows, columns } = imageFrame;
+  if (imageFrame.photometricInterpretation === 'YBR_FULL_422') {
+    return imageFrame.pixelDataLength === 2 * rows * columns && rows % 2 === 0;
+  } else {
```

Comment:
Add a settable flag allowing coercion to RGB if the response is uncompressed, and the original is YBR of some flavour.  That flag could go in cornerstone settings, and maybe be triggered based on the URL prefix (eg url.startsWith), or a regular expression match.

Reply:
Can you please add more to this? I don't understand

Reply:
If the photometric interpretation is YBR_FULL_422, BUT the length is 3*rows*columns, then assume it is the configurable value of either RGB or YBR_FULL.  That is invalid data, but is common enough that we see it regularly as a "bug" fix request.
To decide whether RGB or YBR_FULL, check the datasource configuration - something like:
UncompressedColorEncoding=YBR_FULL | RGB  (default is RGB)
and perhaps
ForceColorEncoding=AllUncompressed | All | WrongLength  (default is WrongLength)

Then, have the above function return the PMI to use for the given image.   

const {ForceColorEncoding = WrongLength, UncompressedColorEncoding = RGB} = dataSourceConfiguration;

if( pixel length === 3 * rows * columns && pmi ==='YBR_FULL_422 && forceColorEncoding===WrongLength ) {
   return UncompressedColorEncoding

indicating that this image needs decompression handling, in the specified color encoding.